pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1884,6 +1884,86 @@ status:
 	}
 }
 
+func TestReconcilePodFailures(t *testing.T) {
+	type testCase struct {
+		name           string
+		taskRun        *v1beta1.TaskRun
+		expectedStatus *apis.Condition
+		wantEvents     []string
+	}
+
+	testcases := []testCase{
+		{
+			name: "taskrun with pod image pull failure",
+			taskRun: parse.MustParseTaskRun(t, `
+metadata:
+  name: test-imagepull-fail
+  namespace: foo
+spec:
+  taskSpec:
+    steps:
+    - image: whatever
+      name: ""
+      resources: {}
+      script: "true"
+  timeout: 1h0m0s
+status:
+  steps:
+  - container: step-unnamed-0
+    name: unnamed-0
+    waiting:
+      message: Back-off pulling image "whatever"
+      reason: ImagePullBackOff
+  taskSpec:
+    steps:
+    - image: whatever
+      name: ""
+      resources: {}
+      script: "true"
```

Comment:
```suggestion
spec:
  taskSpec:
    steps:
    - image: whatever
status:
  steps:
  - container: step-unnamed-0
    name: unnamed-0
    waiting:
      message: Back-off pulling image "whatever"
      reason: ImagePullBackOff
  taskSpec:
    steps:
    - image: whatever
```

In addition to being shorter, this also makes it clearer which fields from the input are actually used during reconciliation.